private/buf/bufsync/syncer.go

```diff
@@ -666,140 +605,70 @@ func (s *syncer) readModuleAt(
 // after those commits were synced.
 func (s *syncer) backfillTags(
 	ctx context.Context,
+	moduleDir string,
+	moduleIdentity bufmoduleref.ModuleIdentity,
 	branch string,
-	syncableCommits []*syncableCommit,
+	syncStartHash git.Hash,
 	clock clock,
 ) error {
-	branchModulesForSync, ok := s.branchesToModulesForSync[branch]
-	if !ok || len(branchModulesForSync) == 0 {
-		// branch should not be synced, or no modules to sync in that branch
-		return nil
-	}
-	moduleToStartPoint, err := s.lookbackStartingPoints(ctx, branch, syncableCommits)
-	if err != nil {
-		return fmt.Errorf("find lookback starting points for branch %s: %w", branch, err)
-	}
-	timeLimit := clock.Now().Add(-lookbackTimeLimit)
-	stopLoopErr := errors.New("stop loop")
-	for moduleDir, startPoint := range moduleToStartPoint {
```

Comment:
No need for this outer loop. The `backfillTags` func is not invoked once per module directory.